datafusion/sql/src/unparser/utils.rs

```diff
@@ -318,7 +318,9 @@ pub(crate) fn try_transform_to_simple_table_scan_with_filters(
                 plan_stack.push(alias.input.as_ref());
             }
             LogicalPlan::Filter(filter) => {
-                filters.push(filter.predicate.clone());
+                if !filters.contains(&filter.predicate) {
```

Comment:
Can we change them to HashSet? including `table_scan_filters`

Reply:
Yeah my previous implementation is actually Hashset, however it doesn't preserve the filter order, so I didn't end up using Hashset

Reply:
Another thing I could do is maintain a temporary Hashset to keep track of the exsiting filters, while the results is still constructed with the Vector. Every filter will be checked if it exists with Hashset before pushing to the Vector. Would this approach be better than just using Vector.contains?

Reply:
How about `IndexSet`?

Reply:
IndexSet would be better so that the order of the filters is preserved